sdk/client-core/README.md

```diff
@@ -0,0 +1,32 @@
+# Azure Core shared library for Java
```

Comment:
This README has a lot of references to Azure. We should clean this up to make in unbranded.